src/app/sidebar/aichat.tsx

```diff
@@ -0,0 +1,322 @@
+                    </>
+                );
+            }
+        }
+
+        const cssVar = GlobalModel.isDev ? "--app-panel-bg-color-dev" : "--app-panel-bg-color";
+        const panelBgColor = getComputedStyle(document.documentElement).getPropertyValue(cssVar).trim();
+        const color = tinycolor(panelBgColor);
+        const newColor = color.isValid() ? tinycolor(panelBgColor).darken(6).toString() : "none";
+        const backgroundColor = this.chatListKeyCount % 2 === 0 ? "none" : newColor;
+
+        return (
+            <div className={msgClassName} key={curKey} style={{ backgroundColor }}>
+                {innerHTML}
+            </div>
+        );
+    }
+
+    render() {
+        const chatMessageItems = GlobalModel.inputModel.AICmdInfoChatItems.slice();
+        const chitem: OpenAICmdInfoChatMessageType = null;
+        return (
+            <OverlayScrollbarsComponent
+                ref={this.containerRef}
+                className="content"
+                options={{ scrollbars: { autoHide: "leave" } }}
+                events={{ initialized: this.onScrollbarInitialized }}
+            >
+                <div ref={this.chatWindowRef} className="chat-window">
+                    <div className="filler"></div>
+                    <For each="chitem" index="idx" of={chatMessageItems}>
+                        {this.renderChatMessage(chitem)}
+                    </For>
+                </div>
+            </OverlayScrollbarsComponent>
+        );
+    }
+}
+
+@mobxReact.observer
+class AIChat extends React.Component<{}, {}> {
+    textAreaRef: React.RefObject<HTMLTextAreaElement> = React.createRef();
+    termFontSize: number = 14;
+
+    componentDidMount() {
+        const inputModel = GlobalModel.inputModel;
+        if (this.textAreaRef.current != null) {
+            this.textAreaRef.current.focus();
+            // inputModel.setCmdInfoChatRefs(this.textAreaRef, this.chatWindowScrollRef);
+        }
+        this.requestChatUpdate();
+        this.onTextAreaChange(null);
+    }
+
+    requestChatUpdate() {
+        const chatMessageItems = GlobalModel.inputModel.AICmdInfoChatItems.slice();
+        if (chatMessageItems == null || chatMessageItems.length == 0) {
+            this.submitChatMessage("");
+        }
+    }
+
+    submitChatMessage(messageStr: string) {
+        const curLine = GlobalModel.inputModel.curLine;
+        const prtn = GlobalModel.submitChatInfoCommand(messageStr, curLine, false);
+        prtn.then((rtn) => {
+            if (!rtn.success) {
+                console.log("submit chat command error: " + rtn.error);
+            }
+        }).catch((_) => {});
+    }
+
+    getLinePos(elem: any): { numLines: number; linePos: number } {
+        const numLines = elem.value.split("\n").length;
+        const linePos = elem.value.substr(0, elem.selectionStart).split("\n").length;
+        return { numLines, linePos };
+    }
+
+    @mobx.action.bound
```

Comment:
you'll need to add a constructor that calls `mobx.makeObservable(this)` in order for these attributes to work